### ISSUE TITLE
feature request: Add pre/post extension hooks.

### ISSUE THREAD

**Issue by jkwmoore** [easybuilders/easybuild-framework#4191]
## Problem description

Currently it seems like the existing hooks cannot be used to amend configuration on extensions within an Easyconfig, e.g. for amending Perl extension `preconfigopts`.

I am currently wanting to override some settings on the SSLeay extension for Perl Easyconfigs which I don't think is directly achievable. I'm attempting to force dependency on base OS OpenSSL libraries which are versioned below the Easybuild OpenSSL configs for various toolchains (Centos 7 with no approval for EPEL OpenSSL v1.1 :cry: )

## Proposed solution

Add some hooks which execute in the post and pre context of extension installations.

## tl;dr

Please add pre/post extension hooks.

**Comment by boegel** [comment]
implemented in #4193
